// react-templates/src/wfs.js
'use strict';

var mkdirp = require('mkdirp');
var fs = require('fs');
var path = require('path');

/**
 * @param {string} file
 * @param {string} content
 */
function writeFileSync(file, content) {
    var newDir = path.dirname(file);
    mkdirp.sync(newDir);
    fs.writeFileSync(file, content);
}

/**
 * @param {string} file
 * @return {string}
 */
function readFileSync(file) {
    return fs.readFileSync(file, 'utf8');
}

/**
 * @param {string} file
 * @return {string}
 */
function normalizeSep(file) {
    return file.replace(/\\/g, '/');
}

/**
 * @param {string} file
 * @return {*}
 */
function readJSON(file) {
    var json = readFileSync(file);
    return JSON.parse(json);
}

/**
 * return basename without extension
 * @param filePath
 * @return {*}
 */
function baseFileName(filePath) {
    return path.basename(filePath, path.extname(filePath));
}

module.exports = {
    file: {
        read: readFileSync,
        write: writeFileSync
    },
    writeFileSync: writeFileSync,
    readFileSync: readFileSync,
    normalizeSep: normalizeSep,
    readJSON: readJSON,
    baseFileName: baseFileName
};
